distributed/system_monitor.py

```diff
@@ -65,7 +65,8 @@ def __init__(
             try:
                 disk_ioc = psutil.disk_io_counters()
             except Exception:
-                # FIXME is this possible?
+                # FIXME occurs when psutil version doesn't have handling for given platform / kernel;
+                # should we explicitly error in this case?
                 monitor_disk_io = False  # pragma: nocover
```

Comment:
Digging into the failures on `distributed/tests/test_system_monitor.py::test_disk_config`, it looks like this exception gets raised when `psutil` doesn't support the given platform/kernel (in our case, 5.6.3 didn't have support for Linux 5.5+).

Given this knowledge, would it make sense to raise a warning here to let the user know that I/O monitoring will be disabled because of this?

Reply:
Nice detective work! I could definitely see how emitting a warning here could help avoid user confusion. Normally, I'd defer to you as to if you'd like to include that new warning/test here or in a separate PR. But given the diff here is already pretty big, and adding a new warning seems more like an (reasonable) enhancement rather than directly related to the mindeps CI build, how about we tackle it in a separate PR?